src/findlib.ml

```diff
@@ -438,12 +438,19 @@ let available t name = Result.is_ok (find t name)
 let root_packages t =
   let pkgs =
     List.concat_map t.paths ~f:(fun dir ->
-      Path.readdir_unsorted dir
-      |> List.filter_map ~f:(fun name ->
-        if Path.exists (Path.relative dir (name ^ "/META")) then
-          Some (Lib_name.of_string_exn ~loc:None name)
-        else
-          None))
+      match Path.readdir_unsorted dir with
+      | Error unix_error ->
+        Errors.warn Loc.none
+          "Unable to read directory %s for findlib package@.Reason:%s@."
+          (Path.to_string_maybe_quoted dir)
+          (Unix.error_message unix_error);
+        []
```

Comment:
I'm not sure how I feel about this general approach of turning errors into warnings. For an interactive session it seems fine, but for unattended builds it could be a hazard where it causes you to build a configuration different from what you expected.

I guess we already have a few places that do this so this change is not making things significantly different.

Reply:
I can undo this one l, but it seems like the findlib dirs can be outside the user's control so it makes sense to at least have this as an option. I can remove this change if there's an objection.

Reply:
I don't understand the concern enough to agree or disagree. I'll leave it to your judgement.

Reply:
I ended up getting rid of it. Let's wait for a user to request this first.